src/jit/optimizer.cpp

```diff
@@ -4648,7 +4648,8 @@ void Compiler::optCloneLoop(unsigned loopInd, LoopCloneContext* context)
         BasicBlock* newBlk = fgNewBBafter(blk->bbJumpKind, newPred,
                                           /*extendRegion*/ true);
 
-        BasicBlock::CloneBlockState(this, newBlk, blk);
+        bool cloneOk = BasicBlock::CloneBlockState(this, newBlk, blk);
+        noway_assert(cloneOk);
```

Comment:
This noway_assert still mean that when we fail to full clone the we will drop to MinOpts.
You should add a new ToDo comment say that we should add some new backout code to handle this case.
I must have run into this case because you changed CloneBlockState so that it can return false.